src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+
+  let stateCopy = { ...state };
+
+  for (const action of actions) {
+    const { type, extraData, keysToRemove } = action;
+
+    switch (type) {
+      case 'addProperties':
+        stateCopy = { ...stateCopy, ...extraData };
+        break;
+
+      case 'removeProperties':
+        for (const item of keysToRemove) {
+          delete stateCopy[item];
+        }
+        break;
+
+      case 'clear':
+        for (const key in stateCopy) {
+          delete stateCopy[key];
+        }
+        break;
+    }
+
+    result.push({ ...stateCopy });
```

Comment:
Instead of creating a copy of stateCopy after each action, consider creating the copy at the beginning of each iteration of the for loop. This way, you avoid mutating the same object and ensure that the original state object is not modified.